ironic/drivers/modules/drac/management.py

```diff
@@ -183,9 +185,13 @@ def set_boot_device(node, device, persistent=False):
     :raises: DracOperationError on an error from python-dracclient.
     """
 
-    drac_job.validate_job_queue(node)
-
     client = drac_common.get_drac_client(node)
+    if client.is_lifecycle_in_recovery():
+        node.driver_internal_info['management_config_substep'] = 'completed'
```

Comment:
It looks to me like this code will make the cleaning step never run.  I think this is the flow that will happen:
set_boot_device is called to set the server to PXE boot 1 time
This code sets the cleaning step to completed
The server will not power cycle or PXE boot
The periodic job will trigger and mark cleaning as done